OneSignalSDK/unittest/src/test/java/com/test/onesignal/MainOneSignalClassRunner.java

```diff
@@ -2448,41 +2473,42 @@ public void ensureNoConcurrentUpdateCallsWithSameData_forPendingActiveTime() thr
       assertEquals(3, ShadowOneSignalRestClient.requests.size());
    }
 
-   @Test
-   public void testMethodCallsBeforeInit() throws Exception {
-      GetTags();
-      OneSignal.sendTag("key", "value");
-      OneSignal.sendTags("{\"key\": \"value\"}");
-      OneSignal.deleteTag("key");
-      OneSignal.deleteTags("[\"key1\", \"key2\"]");
-      OneSignal.setSubscription(false);
-      OneSignal.enableVibrate(false);
-      OneSignal.enableSound(false);
-      OneSignal.promptLocation();
-      OneSignal.postNotification("{}", new OneSignal.PostNotificationResponseHandler() {
-         @Override
-         public void onSuccess(JSONObject response) {}
-         @Override
-         public void onFailure(JSONObject response) {}
-      });
-
-      OneSignal.setInFocusDisplaying(OneSignal.OSInFocusDisplayOption.Notification);
-      OneSignal.removeNotificationOpenedHandler();
-      OneSignal.removeNotificationReceivedHandler();
-
-      OneSignal.startInit(blankActivity).init();
-
-      // Checks that Notification setting worked.
-      Field OneSignal_mInitBuilder = OneSignal.class.getDeclaredField("mInitBuilder");
-      OneSignal_mInitBuilder.setAccessible(true);
-      OneSignal.Builder builder = (OneSignal.Builder)OneSignal_mInitBuilder.get(null);
-      Field OneSignal_Builder_mDisplayOption = builder.getClass().getDeclaredField("mDisplayOption");
-      OneSignal_Builder_mDisplayOption.setAccessible(true);
-      OneSignal.OSInFocusDisplayOption inFocusDisplayOption = (OneSignal.OSInFocusDisplayOption)OneSignal_Builder_mDisplayOption.get(builder);
-      assertEquals(inFocusDisplayOption, OneSignal.OSInFocusDisplayOption.Notification);
-
-      threadAndTaskWait();
-   }
+   // TODO: Builder method should be deleted?
+//   @Test
+//   public void testMethodCallsBeforeInit() throws Exception {
```

Comment:
We need to keep keep the intent of this test. We should maybe split this into 3 tests.
1. Methods calls before setAppId and setContext
2. Methods calls before setAppId
3. Methods calls before setContext

Reply:
Okay sounds good to me